internal/store/memory/store.go

```diff
@@ -58,22 +57,6 @@ func (s *Store) GatewayExists(ctx context.Context, id core.GatewayID) (bool, err
 	return found, nil
 }
 
-func (s *Store) CanFetchSecrets(ctx context.Context, id core.GatewayID, secrets []string) (bool, error) {
```

Comment:
This moved onto the `store.Gateway` interface to limit the amount of logic the store has that isn't related to storing things